torch/csrc/api/include/torch/nn/modules/transformercoder.h

```diff
@@ -85,47 +85,48 @@ TORCH_MODULE(TransformerEncoder);
 /// auto out = transformer_decoder(tgt, memory);
 /// ```
 class TORCH_API TransformerDecoderImpl : public Cloneable<TransformerDecoderImpl> {
- public:
-  TransformerDecoderImpl(TransformerDecoderLayer decoder_layer, int64_t num_layers)
-    : TransformerDecoderImpl(TransformerDecoderOptions(decoder_layer, num_layers)) {}
-  explicit TransformerDecoderImpl(TransformerDecoderOptions options_);
-
-  void reset() override;
-
-  void reset_parameters();
-
-  /// Pass the inputs (and mask) through the decoder layer in turn.
-  ///Args:
-  ///       tgt: the sequence to the decoder layer (required).
-  ///       memory: the sequence from the last layer of the encoder (required).
-  ///       tgt_mask: the mask for the tgt sequence (optional).
-  ///       memory_mask: the mask for the memory sequence (optional).
-  ///       tgt_key_padding_mask: the mask for the tgt keys per batch (optional).
-  ///       memory_key_padding_mask: the mask for the memory keys per batch (optional).
-  Tensor forward(const Tensor& tgt,
-                 const Tensor& memory,
-                 const Tensor& tgt_mask = {},
-                 const Tensor& memory_mask = {},
-                 const Tensor& tgt_key_padding_mask = {},
-                 const Tensor& memory_key_padding_mask = {});
-
-  /// The options used to configure this module.
-  TransformerDecoderOptions options;
-
-  ///Cloned layers of decoder layers
-  ModuleList layers{nullptr};
-
-  ///optional layer normalization module
-  AnyModule norm;
-
- protected:
-  FORWARD_HAS_DEFAULT_ARGS(
+
+  public:
+    TransformerDecoderImpl(TransformerDecoderLayer decoder_layer, int64_t num_layers)
+      : TransformerDecoderImpl(TransformerDecoderOptions(decoder_layer, num_layers)) {}
+    explicit TransformerDecoderImpl(TransformerDecoderOptions options_);
+
+    void reset() override;
+
+    void reset_parameters();
```

Comment:
Are the spacing changes intentional? Do we have a PyTorch C++ API style guide?

Reply:
reset_parameters() has no regulation. Since this is not required for every module and basically a customized function. I am trying to make it always named reset_parameters(). I will enter another PR to change all _reset_parameters(), reset_parameters_() into reset_parameter(). 
We used to have a ticket to make this function virtual. But we did not proceed since this is not needed by every module and making function virtual will lower the performance.

Reply:
Sorry, my question is that it looks like there has been formatting changes. Previously, there were fewer spaces, now, there are more spaces. Is the addition of spaces intentional?

Reply:
I see. Yes, the decoder part is contributed from external. 
We follow the the convention that with 2 chars ident. For each level, so I corrected it for decoder impl.